piff/gp_interp_2pcf.py

```diff
@@ -0,0 +1,372 @@
+                 logger=None, white_noise=0.):
+
+        self.keys = keys
+        self.npca = npca
+        self.degenerate_points = False
+        self.normalize = normalize
+        self.optimize = optimize
+        self.white_noise = white_noise
+
+        self.kwargs = {
+            'keys': keys,
+            'optimize': optimize,
+            'npca': npca,
+            'kernel': kernel,
+            'normalize':normalize
+        }
+
+        if len(keys)!=2:
+            raise ValueError('the total size of keys can not be something else than 2 using two-point correlation function. Here len(keys) = %i'%(len(keys)))
+
+        if type(kernel) is str:
+            self.kernel_template = [self._eval_kernel(kernel)]
+        else:
+            if type(kernel) is not list and type(kernel) is not np.ndarray:
+                raise TypeError("kernel should be a string a list or a numpy.ndarray of string")
+            else:
+                self.kernel_template = [self._eval_kernel(ker) for ker in kernel]
+
+        self._2pcf = []
+        self._2pcf_dist = []
+        self._2pcf_fit = []
+
+    @staticmethod
+    def _eval_kernel(kernel):
+        # Some import trickery to get all subclasses of sklearn.gaussian_process.kernels.Kernel
+        # into the local namespace without doing "from sklearn.gaussian_process.kernels import *"
+        # and without importing them all manually.
+        def recurse_subclasses(cls):
+            out = []
+            for c in cls.__subclasses__():
+                out.append(c)
+                out.extend(recurse_subclasses(c))
+            return out
+        clses = recurse_subclasses(Kernel)
+        for cls in clses:
+            module = __import__(cls.__module__, globals(), locals(), cls)
+            execstr = "{0} = module.{0}".format(cls.__name__)
+            exec(execstr, globals(), locals())
+
+        try:
+            k = eval(kernel)
+        except:
+            raise RuntimeError("Failed to evaluate kernel string {0!r}".format(kernel))
+        
+        if type(k.theta) is property:
+            raise TypeError("String provided was not initialized properly")
+        return k
+
+    def _fit(self, kernel, X, y, y_err=None, logger=None):
+        """Update the Kernel with data.
+
+        :param kernel: sklearn.gaussian_process kernel.
+        :param X:  The independent covariates.  (n_samples, 2)
+        :param y:  The dependent responses.  (n_samples, n_targets)
+        :param y_err: Error of y. (n_samples, n_targets) [default: None]
+        :param logger:  A logger object for logging debug info. [default: None]
+        """
+        if logger:
+            logger.debug('Start kernel: %s', kernel.set_params())
+            logger.debug('gp.fit with mean y = %s',np.mean(y))
+        # Save these for potential read/write.                
+        if y_err is None:
+            y_err = np.zeros_like(y)
```

Comment:
This branch is never tested.  Nor the other functions where you allow y_err to be None.  I think that given how `solve` is written, it's not possible to get y_err=None.  So probably remove that default and option.  Else add a test that uses `y_err=None`.